akka-http/shared/src/main/scala/org/lyranthe/prometheus/client/integration/akkahttp/Metrics.scala

```diff
@@ -0,0 +1,85 @@
+package org.lyranthe.prometheus.client.integration.akkahttp
+
+import akka.http.scaladsl.server.{Directive, Directive0, Directives, Route}
+import org.lyranthe.prometheus.client._
+
+trait Metrics {
+
+  /**
+    * Returns the Akka Http Route to expose the metrics
+    */
+  val routes: Route
+
+  /**
+    * Records metrics for a given endpoint.
+    */
+  def withMetrics(endpoint: Option[String] = None): Directive0
+}
+
+trait PrometheusRoutes extends Directives {
+
+  val metricsPathDirective: Directive[Unit]
+
+  implicit val registry: Registry
+
+  val routes: Route = metricsPathDirective {
+    get {
+      complete(registry.outputText)
+    }
+  }
+
+}
+
+/**
+  * Creates a PrometheusMetrics with a Routing directive to expose metrics and a custom DSL directive to monitor endpoints.
+  *
+  * @param metricsPathDirective the path directive to expose the metrics (default to /metrics)
+  * @param enableJMX if true expose jms statistics (default to true)
+  * @param registry the Prometheus Client Registry service
+  */
+class PrometheusMetrics(override val metricsPathDirective: Directive[Unit] = Directives.path("metrics"),
+                        enableJMX: Boolean = true)(implicit val registry: Registry)
+  extends Metrics with PrometheusRoutes {
+
+  if (enableJMX) {
+    jmx.register()
+  }
+
+  /**
+    *
+    * WARNING: every recorded endpoint stores a new time series in database. Do not record variable path parameters (eg: /brands/123,
+    * /brands/456) or query params to avoid to dramatically increase the amount of data stored. When None is used as endpoint make
+    * sure to position the directive at a level where the matched path cannot produce unbounded sets of values.
+    *
+    * @param endpoint if None uses [[akka.http.scaladsl.server.Directives.extractMatchedPath]]
+    */
+  def withMetrics(endpoint: Option[String] = None): Directive0 = extractRequestContext.flatMap { ctx =>
+    val timer = Timer()
+    extractMatchedPath.flatMap { path =>
+      mapResponse { response =>
+        val label = endpoint.getOrElse(path.toString())
```

Comment:
IMHO I would prefer to keep the API as it is and document it properly (there is a warning in the scaladoc already). If there are no path parameters this is completely safe. Otherwise is up to the user to specify it. @fiadliel WDYT?